packages/protocol-kit/src/SafeProvider.ts

```diff
@@ -246,34 +305,46 @@ class SafeProvider {
 
     if (isTypedDataSigner(signer)) {
       const typedData = generateTypedData(safeEIP712Args)
-      const signature = await signer.signTypedData(
-        typedData.domain,
-        typedData.primaryType === 'SafeMessage'
-          ? { SafeMessage: (typedData as EIP712TypedDataMessage).types.SafeMessage }
-          : { SafeTx: (typedData as EIP712TypedDataTx).types.SafeTx },
-        typedData.message
-      )
+      const { chainId, verifyingContract } = typedData.domain
+      const chain = chainId ? Number(chainId) : undefined // ensure empty string becomes undefined
+      const domain = { verifyingContract: asAddress(verifyingContract), chainId: chain }
+
+      const signature = await signer.signTypedData({
+        domain,
+        types:
+          typedData.primaryType === 'SafeMessage'
+            ? { SafeMessage: (typedData as EIP712TypedDataMessage).types.SafeMessage }
+            : { SafeTx: (typedData as EIP712TypedDataTx).types.SafeTx },
+        primaryType: typedData.primaryType,
+        message: typedData.message
+      })
       return signature
     }
 
     throw new Error('The current signer does not implement EIP-712 to sign typed data')
   }
 
   async estimateGas(transaction: SafeProviderTransaction): Promise<string> {
-    return (await this.#externalProvider.estimateGas(transaction)).toString()
+    const anyTransaction = transaction as any
```

Comment:
Does it work with `as any`? Is there no conversion needed for some transaction props, e.g. from string to bigint?